app/src/main/java/org/simple/clinic/allpatientsinfacility/AllPatientsInFacilityListItem.kt

```diff
@@ -0,0 +1,70 @@
+package org.simple.clinic.allpatientsinfacility
+
+import android.annotation.SuppressLint
+import androidx.recyclerview.widget.DiffUtil
+import kotlinx.android.synthetic.main.list_allpatientsinfacility_facility_header.*
+import kotlinx.android.synthetic.main.list_patient_search.*
+import org.simple.clinic.R
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.patient.PatientSearchResult
+import org.simple.clinic.widgets.recyclerview.ViewHolderX
+
+sealed class AllPatientsInFacilityListItem {
+
+  companion object {
+    fun mapSearchResultsToListItems(
```

Comment:
This is logic that should be part of `AllPatientsInFacilityUiChangeProducer` isn't it?

Reply:
@ragunathjawahar and I had a discussion about whether this conversion of domain models to view models should happen inside the `UiChangeProducer` or the `View`. The outcome of this discussion was that since this is a presentation detail (that the domain models are being converted into list items which is specific to `RecyclerView`), they should not happen here and should be handled by the view itself.

That way, the presentation of these items can change without affecting the ui change producer. Does this make sense?